ui/server.py

```diff
@@ -316,18 +318,63 @@ def get_repos():
             if repo['url'] == metadata[0]:
                 repo['total'] = metadata[1]  # Total number of discoveries
                 repo['TP'] = metadata[2]  # Number of new discoveries
+                repo['FP'] = metadata[3]  # Number of true positive discoveries
+                # Number of addressing discoveries
+                repo['addressing'] = metadata[4]
+                # Number of irrelevant discoveries
+                repo['not_relevant'] = metadata[5]
+                repo['fixed'] = metadata[6]  # Number of fixed discoveries
                 break  # We found the repo, no need to check next metadata
             else:
                 # The repo was scanned and doesn't have discoveries
                 repo['total'] = 0
                 repo['TP'] = 0
+                repo['FP'] = 0
+                repo['addressing'] = 0
+                repo['not_relevant'] = 0
+                repo['fixed'] = 0
+
         repo['scan_active'] = False
         if repo['url'] in active_scans:
             repo['scan_active'] = True
 
     return jsonify(repos)
 
 
+@app.route('/export_discoveries_csv', methods=['GET', 'POST'])
+def export_discoveries_csv():
+    url = request.form.get('repo_url')
+    _, discoveries = c.get_discoveries(url)
+
+    states = []
+    if request.form.get('checkAll') == 'all':
+        states = ['new', 'false_positive',
+                  'addressing', 'not_relevant', 'fixed']
+    else:
+        states = request.form.getlist('check')
+
+    filtered_discoveries = list(
+        filter(lambda d: d.get('state') in states, discoveries))
+
+    try:
+        stringIO = io.StringIO()
+        csv_writer = csv.DictWriter(stringIO, discoveries[0].keys())
```

Comment:
do you get an IndexError if you try to export discoveries for a repo that didn't have any?

Reply:
Yes.